### PR TITLE
Version 3

### PR DESCRIPTION
**Pull request by Derisis13**
# Pull Request Template

## Type of change

- [ ] Bug fix
- [x] Feature
- [x] Refactor / QA / enhancement
- [x] Documentation update

## Description

V3 wanted features:
 - [x] modular UI 
 - [ ] UI documentation
 - [x] modular player backend 
 - [ ] player backend documentation
 - [x] provider change (animixplay?)
 - [ ] Cool logo (?)

## Checklist

- [x] any anime playing
- [x] bumped version
- [x] next, prev and replay work
- [x] quality works
- [x] downloads work
- [x] quality works with downloads
- [x] select episode -a and rapid resume work
- [ ] syncplay -s works for old and new anime
- [x] autoplay, aka range selection, works

## Additional Testcases

- The safe bet: One Piece
- Episode 0: Saenai Heroine no Sodatekata ♭
- Unicode: Saenai Heroine no Sodatekata ♭
- Not uploaded: one piece dub episode 590
- Unreleased: soredemo ayumu wa yosetekuru
- Short id (for decryption): Log Horizon episode 1-2

### CODE REVIEW

**Comment by Derisis13** [comment]
Yep, this is bloated... Will work on that later.

File: ani-cli [hunk 1]
```diff
@@ -158,61 +152,49 @@ search_history () {
...
-	ep_no="$2"
-	curl -A "$agent" -s "$base_url/videos/${anime_id}${ep_no}" | sed -nE 's_^[[:space:]]*<iframe src="([^"]*)".*_\1_p' |
-		sed 's/^/https:/g'
-}
-
 generate_link() {
 	case $1 in
 		1)
 			refr=$(printf "%s" "$links" | grep "mp4upload")
+			[ -z "$refr" ] && return 0
```
Comment (Derisis13) [inline ani-cli:159]: return 0 makes the testing on the next line redundant

File: ani-cli [hunk 2]
```diff
@@ -158,61 +152,49 @@ search_history () {
...
 			result_links="$(printf '%s' "$data" | base64 -d 2>/dev/null | openssl enc -d -aes256 -K "$second_key" -iv "$iv" 2>/dev/null | sed -e 's/\].*/\]/' -e 's/\\//g' | grep -Eo 'https:\/\/[-a-zA-Z0-9@:%._\+~#=][a-zA-Z0-9][-a-zA-Z0-9@:%_\+.~#?&\/\/=]*')";;
 	esac
 }
 
 # chooses the link for the set quality
 get_video_link() {
 	dpage_url="$1"
-	resp=$(curl -A "$agent" -s "$1")
+	id=$(printf "%s" "$dpage_url" | sed -nE 's/.*id=(.*)&title.*/\1/p')
+	resp="$(curl -A "$agent" -s "https://goload.pro/streaming.php?id=$id" | sed -nE 's/.*class="container-(.*)">/\1/p ; s/.*class="wrapper container-(.*)">/\1/p ; s/.*class=".*videocontent-(.*)">/\1/p ; s/.*data-value="(.*)">.*/\1/p ; s/.*data-status="1".*data-video="(.*)">.*/\1/p')"
```
Comment (Derisis13) [inline ani-cli:190]: this is an ugly long line, should be a function instead (all the things in the curllll)

Comment (Derisis13) [reply ani-cli:190]: Also is this the only place where we use `goload.pro`? If not, it should be extracted to a variable, as it'll change sooner or later...

Comment (CoolnsX) [reply ani-cli:197]: > this is an ugly long line, should be a function instead (all the things in the subshell)

okay.. will do it in future..

Comment (CoolnsX) [reply ani-cli:197]: > Also is this the only place where we use `goload.pro`? If not, it should be extracted to a variable, as it'll change sooner or later...

yes.. for extracting keys and ivs and alternate server links of selected episode is done by this link.. i concatenate all sed to make debugging easy..

Comment (Derisis13) [reply ani-cli:197]: Oh I didn't realize it was one sed command... I thought it was a series of commands because of all the semicolons...

**Comment by Derisis13** [comment]
@CoolnsX is history broken only on my part?

**Comment by port19x** [comment]
I'm now using this as my version of ani-cli and will report any issues should I run into them.
I'll do more thorough testing and review once you're ready

**Comment by Derisis13** [comment]
From my end only the documentation remains and I'm good to go

**Comment by Derisis13** [comment]
Oh, and I'll have to reverse the hist_transition script probably as well

**Review by port19x — APPROVED** [review]

**Pull request merged**